src/main/java/org/gwtproject/tools/cldr/CurrencyListProcessor.java

```diff
@@ -0,0 +1,366 @@
+        pw.println("# If # of decimal digits is omitted, 2 is used");
+        pw.println("# If a currency is not generally used, not-used-flag=1");
+        pw.println("# If a currency should be rounded to a multiple of of the least significant");
+        pw.println("#   digit, rounding will be present");
+        pw.println("# Trailing empty fields can be omitted");
+        pw.println();
+    }
+
+    @Override
+    protected void writeOutputFiles() throws IOException {
+        Set<GwtLocale> localesToWrite = localeData.getNonEmptyLocales();
+        String path = "shared/cldr/impl/";
+        String packageName = "org.gwtproject.i18n.shared.cldr.impl";
+        Map<GwtLocale, List<LocaleData.Currency>> temp = this.localesCurrencies;
+        Map<String, CurrencyInfo> allCurrencyData = new HashMap<>();
+        String lastDefaultCurrencyCode = null;
+
+        List<GwtLocale> sorted = localesToWrite.stream()
+                .sorted(Comparator.comparing(GwtLocale::getAsString)
+                        .reversed())
+                .collect(Collectors.toList());
+
+        generateFactory(path, packageName, sorted);
+
+        for (GwtLocale locale : sorted) {
+            Map<String, String> currencyData = localeData.getEntries("currency", locale);
+            String[] currencies = new String[currencyData.size()];
+            currencyData.keySet().toArray(currencies);
+            Arrays.sort(currencies);
+
+            try {
+
+                Properties extraInfo = new Properties();
+                InputStream extraResource = this.getClass().getClassLoader().getResourceAsStream("CurrencyExtra" + Processor.localeSuffix(locale) + ".extraInfo");
+                if (nonNull(extraResource)) {
+                    extraInfo.load(extraResource);
+                }
+
+                for (String currencyCode : currencies) {
+                    CurrencyInfo currencyInfo = new CurrencyInfo(currencyCode, currencyData.get(currencyCode), extraInfo.getProperty(currencyCode));
+                    allCurrencyData.put(currencyCode, currencyInfo);
+                }
+
+                String defCurrencyCode = getLocaleDefaultCurrency(locale);
+                // If this locale specifies a particular locale, or the one that is
+                // inherited has been changed in this locale, re-specify the default
+                // currency so the method will be generated.
+                if (defCurrencyCode == null && currencyData.containsKey(lastDefaultCurrencyCode)) {
+                    defCurrencyCode = lastDefaultCurrencyCode;
+                }
+
+                if (!currencyData.isEmpty() || defCurrencyCode != null) {
+                    generateOnLocale(path, packageName, locale, currencies, allCurrencyData, defCurrencyCode);
+                    lastDefaultCurrencyCode = defCurrencyCode;
+                }
+            } catch (Exception ex) {
+                LOGGER.log(Level.SEVERE, "failed to generate currency list : ", ex);
```

Comment:
do we need to re-throw or something? would hate to have this just get lost in the noise of all the other logging going on...

Reply:
Yes you are right.